services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/compose.py

```diff
@@ -0,0 +1,177 @@
+import logging
```

Comment:
basically  this module define API entrypoints for the ``compose`` resource. 
Essentially, you wrap the functionality of ``docker-compose`` in the following handlers
1. CREATE  ``docker-compose.yaml``
2. ``docker-compose up ... --no-start``
3. ``docker-compose up ... --detach``
4. ``docker-compose up ... pull``
5. ``docker-compose stop ...``
6. DELETE docker-compose.yaml

in view of the above, and from the point of view of a resource-based web API:
1. what is a ``compose`` resource?
2. which are the standard CRUD operations?
3. could we separate the configuration into a sub-resource?

I think there is a lot of room to simplify the design. we can chat separately.

Reply:
Essentially I wrap the lifecycle of using the `docker-compose` command. The idea behind it is the following:
- store the docker-compose.yaml (also applies some minimum validation and ensures to avoid naming collisions for container names an network names)
- pulls images via `docker-compose -p {project} -f {file_path} pull --include-deps`
- start containers via `docker-compose -p {project} -f {file_path} up --no-build -d`
- when the service needs to be stopped `docker-compose -p {project} -f {file_path} stop -t {stop_and_remove_timeout}` will be issued with a 5 second timeout to finish all containers after which they will be killed ensuring everything is cleaned up before the dynamic-service gets removed

Please note that some entrypoints have been added but I am not sure if they will be used. Will cleanup them properly when integrating with https://github.com/ITISFoundation/osparc-simcore/pull/1887